hash_practice/exercises.py

```diff
@@ -2,18 +2,46 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    basket = {} 
+
+    for value in strings: 
+        key = ''.join(sorted(value)) 
+        if key in basket.keys(): 
+            basket[key].append(value) 
+        else: 
+            basket[key] = [] 
+            basket[key].append(value) 
+
+    result = []
+    for key,value in basket.items(): 
+        result.append(value)
+    return result 
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
         Time Complexity: ?
-        Space Complexity: ?
+        Space Complexity: O(n)
     """

```

Comment:
👍  Because of the loop in lines 40-43, the time complexity is O(nk) because you have a loop going k times and each time finding the max element from the frequency map.